### PR TITLE
Query: Migrate insidePolygon to handle several polygons, update insideBoundingBox documentation

### PR DESCRIPTION
**Pull request by PLNech**
Update `{set,get}InsidePolygon` to handle one or several polygons.

### CODE REVIEW

File: algoliasearch/src/main/java/com/algolia/search/saas/Query.java [hunk 1]
```diff
@@ -714,12 +737,69 @@ public int hashCode() {
                                 Double.parseDouble(fields[2 * i + 1])
```
Comment (clement-leprovost) [inline algoliasearch/src/main/java/com/algolia/search/saas/Query.java]: This call can throw a `NumberFormatException` (unchecked exception), therefore `parse(...)` can throw, therefore `getInsidePolygon()` can throw... which is probably not a good idea. :/

File: algoliasearch/src/main/java/com/algolia/search/saas/Query.java [hunk 2]
```diff
@@ -619,11 +619,11 @@ public int hashCode() {
...
      * Search for entries inside a given area defined by the two extreme points
-     * of a rectangle. At indexing, geoloc of an object should be set with
-     * _geoloc attribute containing lat and lng attributes (for example
-     * {"_geoloc":{"lat":48.853409, "lng":2.348800}})
+     * of a rectangle.
      * <p>
-     * You can use several bounding boxes (OR) by calling this method several times.
+     * At indexing, geoloc of an object should be set with
+     * {@code _geoloc} attribute containing lat and lng attributes (for example
+     * {@code {"_geoloc":{"lat":48.853409, "lng":2.348800}})}
```
Comment (clement-leprovost) [inline algoliasearch/src/main/java/com/algolia/search/saas/Query.java]: I would remove those comments. They are legacy from the API Client doc, but we should avoid duplicating this kind of information everywhere; instead, we should have a section (in the API Client doc) explaining how to specify geo information.

File: algoliasearch/src/main/java/com/algolia/search/saas/Query.java [hunk 3]
```diff
@@ -714,12 +737,69 @@ public int hashCode() {
...
+    }
+
+    private static final String KEY_INSIDE_POLYGON = "insidePolygon";
+
+    /**
+     * Search for entries inside a given area defined by the points of a polygon.
+     * <p>
+     * At indexing, geoloc of an object should be set with {@code _geoloc} attribute containing lat
+     * and lng attributes (for example {@code {"_geoloc":{"lat":48.853409, "lng":2.348800}})}
+     */
```
Comment (clement-leprovost) [inline algoliasearch/src/main/java/com/algolia/search/saas/Query.java]: Same here.

File: algoliasearch/src/main/java/com/algolia/search/saas/Query.java [hunk 4]
```diff
@@ -714,12 +737,69 @@ public int hashCode() {
...
+        set(KEY_INSIDE_POLYGON, points == null ? null : new Polygon(points).toString());
+        return this;
+    }
+
+    /**
+     * Search for entries inside a given area defined by several polygons.
+     * <p>
+     * At indexing, geoloc of an object should be set with {@code _geoloc} attribute containing lat
+     * and lng attributes (for example {@code {"_geoloc":{"lat":48.853409, "lng":2.348800}})}
+     */
```
Comment (clement-leprovost) [inline algoliasearch/src/main/java/com/algolia/search/saas/Query.java]: Same here.

File: algoliasearch/src/main/java/com/algolia/search/saas/Query.java (hunk 2 above)
Comment (PLNech) [inline algoliasearch/src/main/java/com/algolia/search/saas/Query.java]: Fixed in 0c4beb9.

File: algoliasearch/src/main/java/com/algolia/search/saas/Query.java (hunk 1 above)
Comment (PLNech) [inline algoliasearch/src/main/java/com/algolia/search/saas/Query.java]: Solved in e4c25c7.

**Pull request merged**